alvr/client_core/src/connection.rs

```diff
@@ -508,14 +482,16 @@ async fn stream_pipeline(
             .await?;
         async move {
             loop {
-                let packet = receiver.recv().await?.header;
-
-                EVENT_QUEUE.lock().push_back(ClientEvent::Haptics {
-                    device_id: packet.path,
-                    duration: packet.duration,
-                    frequency: packet.frequency,
-                    amplitude: packet.amplitude,
-                });
+                let packet = receiver.recv().await?;
+
+                if !packet.had_packet_loss {
+                    EVENT_QUEUE.lock().push_back(ClientEvent::Haptics {
+                        device_id: packet.header.path,
+                        duration: packet.header.duration,
+                        frequency: packet.header.frequency,
+                        amplitude: packet.header.amplitude,
+                    });
+                }
```

Comment:
Why you don't want to send the haptics if there has been a packet loss?

Reply:
Actually it's semantically different now and it means that the current packet is lost and should not be used. I'll update the field name.

Reply:
I thin this is a bad API. the packet should not be accessible if it is corrupted. Either use my previous semantic (the packet lost arrives on the next correct packet) or use an enum `{ PacketReceived { header: T, payload: Bytes }, PacketLost}`.

Reply:
Previous semantic cannot be used technically. When the packet is assembled from shards, it's either packet is complete or not. If I give out incomplete packet and tell that it was actually lost later - that means we'll be using corrupted data, and only after that figure out that it was wrong thing to do.

But how enum is going to be used?